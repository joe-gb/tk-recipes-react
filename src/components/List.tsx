import React from "react";
import styled from 'styled-components';


export const ListItem = styled.div`
  font-size: 18px;
  font-weight: 400;
  line-height: 24px;
  text-align: left;
  vertical-align: baseline;
  text-transform: unset;
  color: #4C5566;
  background-color: #eff2f7;
  margin: 2px 0px 2px 0px;
  padding: 10px 20px;
  display: flex;
  align-items: center;
  justify-content: space-between
  
`;
