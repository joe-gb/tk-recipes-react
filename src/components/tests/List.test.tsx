import React from "react";
import { render, screen } from "@testing-library/react";
import { LinkList, List } from "../List";
import { BrowserRouter as Router } from "react-router-dom";

test("List snapshot test", () => {
  let iterable = ["One", "Two", "Three", "Four", "Five"];
  let { container } = render(<>{List(iterable)}</>);
  expect(container).toMatchSnapshot();
  expect(container.textContent).toBe("OneTwoThreeFourFive");
});

test("LinkList snapshot test", () => {
  let iterable = [
    { link: "/url1/", text: "Link1" },
    { link: "/url2/", text: "Link2" },
  ];
  let { container } = render(<Router>{LinkList(iterable)}</Router>);
  expect(container).toMatchSnapshot();
  expect(container.textContent).toBe("Link1Link2");
});
