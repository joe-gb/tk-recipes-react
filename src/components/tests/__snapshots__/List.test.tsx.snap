// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkList snapshot test 1`] = `
.c0 {
  font-size: 18px;
  font-weight: 400;
  line-height: 24px;
  text-align: left;
  vertical-align: baseline;
  text-transform: unset;
  color: #4c5566;
  background-color: #eff2f7;
  margin: 2px 0px 2px 0px;
  padding: 10px 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div>
  <a
    href="/url1/"
  >
    <div
      class="c0"
    >
      Link1
    </div>
  </a>
  <a
    href="/url2/"
  >
    <div
      class="c0"
    >
      Link2
    </div>
  </a>
</div>
`;

exports[`List snapshot test 1`] = `
.c0 {
  font-size: 18px;
  font-weight: 400;
  line-height: 24px;
  text-align: left;
  vertical-align: baseline;
  text-transform: unset;
  color: #4c5566;
  background-color: #eff2f7;
  margin: 2px 0px 2px 0px;
  padding: 10px 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div>
  <div
    class="c0"
  >
    One
  </div>
  <div
    class="c0"
  >
    Two
  </div>
  <div
    class="c0"
  >
    Three
  </div>
  <div
    class="c0"
  >
    Four
  </div>
  <div
    class="c0"
  >
    Five
  </div>
</div>
`;
